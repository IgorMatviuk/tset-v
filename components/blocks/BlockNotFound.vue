<template>
  <div class="not-found">
    <div class="not-found__background">
      <span class="not-found__error">404</span>
      <picture>
        <source
          data-srcset="/img/error-page/background_xxl.webp"
          type="image/webp"
          media="(min-width: 1500px)"
        />
        <source
          data-srcset="/img/error-page/background_xxl.png"
          type="image/png"
          media="(min-width: 1500px)"
        />
        <source
          data-srcset="/img/error-page/background_xl.webp"
          type="image/webp"
          media="(min-width: 1200px)"
        />
        <source
          data-srcset="/img/error-page/background_xl.png"
          type="image/png"
          media="(min-width: 1200px)"
        />
        <source
          data-srcset="/img/error-page/background_lg.webp"
          type="image/webp"
          media="(min-width: 992px)"
        />
        <source
          data-srcset="/img/error-page/background_lg.png"
          type="image/png"
          media="(min-width: 992px)"
        />
        <source
          data-srcset="/img/error-page/background_md.webp"
          type="image/webp"
          media="(min-width: 768px)"
        />
        <source
          data-srcset="/img/error-page/background_md.png"
          type="image/png"
          media="(min-width: 768px)"
        />
        <source
          data-srcset="/img/error-page/background.webp"
          type="image/webp"
        />
        <img
          data-src="/img/error-page/background.png"
          data-object-fit="cover"
          data-object-position="center center"
          alt="hanger"
          class="not-found__image lazyload"
          width="100"
          height="100"
          loading="lazy"
        />
      </picture>
    </div>
    <BContainer class="not-found__container">
      <div class="not-found__title">{{ $t('errorPage.subtitle') }}</div>
      <div class="not-found__description">
        {{ $t('errorPage.description') }}
      </div>
      <BButton class="not-found__link" :to="$i18n.path('')" variant="primary">
        {{ $t('buttons.toTheMain') }}
      </BButton>
    </BContainer>
  </div>
</template>

<script>
export default {}
</script>

<style lang="scss">
.not-found {
  overflow: hidden;
  background-color: $primary-light-3;

  &__error {
    display: none;
  }

  &__image {
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: center center;
  }

  &__background {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    overflow: hidden;
  }

  &__container {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    min-height: calc(100vh - 80px);
    padding-bottom: 60px;
  }

  &__title {
    margin-bottom: 9px;
    font-size: 20px;
    font-weight: 500;
    line-height: 25px;
    text-align: center;
  }

  &__description {
    max-width: 270px;
    margin-bottom: 26px;
    font-size: 12px;
    line-height: 15px;
    color: $text-color;
    text-align: center;
  }

  @include media-breakpoint-up(md) {
    &__error {
      position: absolute;
      top: 54%;
      left: 50%;
      display: block;
      font-size: 350px;
      font-weight: 700;
      line-height: 440px;
      color: rgba($color: $white, $alpha: 0.3);
      text-transform: uppercase;
      transform: translate(-50%, -50%);
    }

    &__container {
      min-height: calc(100vh - 90px);
    }

    &__title {
      margin-bottom: 10px;
      font-size: 24px;
      line-height: 30px;
    }

    &__description {
      max-width: 320px;
      margin-bottom: 29px;
      font-size: 14px;
      line-height: 18px;
    }
  }

  @include media-breakpoint-up(lg) {
    &__error {
      top: 56%;
    }
  }

  @include media-breakpoint-up(xl) {
    &__error {
      top: 54%;
      font-size: 400px;
      line-height: 503px;
    }
  }

  @include media-breakpoint-up(xxl) {
    &__error {
      top: 51.5%;
      font-size: 500px;
      line-height: 629px;
    }
  }
}
</style>
