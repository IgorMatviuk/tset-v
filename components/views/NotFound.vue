<template>
  <main class="not-found-page">
    <BlockNotFound />
  </main>
</template>

<script>
import BlockNotFound from '@/components/blocks/BlockNotFound'

export default {
  components: {
    BlockNotFound
  }
}
</script>

<style lang="scss">
.not-found-page {
  // position: fixed;
  // top: 0;
  // right: 0;
  // bottom: env(safe-area-inset-bottom);
  // left: 0;
  // padding-top: 45px;
  background-color: $primary-light-3;
}
</style>
