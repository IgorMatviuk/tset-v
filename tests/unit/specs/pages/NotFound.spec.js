import { mount, createLocalVue } from '@vue/test-utils'
import { BContainer, BButton } from 'bootstrap-vue'

import NotFound from '@/components/views/NotFound'
import BlockNotFound from '@/components/blocks/BlockNotFound'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)
localVue.component('BButton', BButton)

describe('NotFound', () => {
  test('renders not found page', () => {
    const wrapper = mount(NotFound, {
      localVue,
      mocks: {
        $t: () => '',
        $i18n: { path: (url) => url }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.findComponent(BlockNotFound).exists()).toBe(true)
  })
})
