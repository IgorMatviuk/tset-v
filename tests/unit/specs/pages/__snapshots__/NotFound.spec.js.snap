// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders not found page 1`] = `
<main
  class="not-found-page"
>
  <div
    class="not-found"
  >
    <div
      class="not-found__background"
    >
      <span
        class="not-found__error"
      >
        404
      </span>
       
      <picture>
        <source
          data-srcset="/img/error-page/background_xxl.webp"
          media="(min-width: 1500px)"
          type="image/webp"
        />
         
        <source
          data-srcset="/img/error-page/background_xxl.png"
          media="(min-width: 1500px)"
          type="image/png"
        />
         
        <source
          data-srcset="/img/error-page/background_xl.webp"
          media="(min-width: 1200px)"
          type="image/webp"
        />
         
        <source
          data-srcset="/img/error-page/background_xl.png"
          media="(min-width: 1200px)"
          type="image/png"
        />
         
        <source
          data-srcset="/img/error-page/background_lg.webp"
          media="(min-width: 992px)"
          type="image/webp"
        />
         
        <source
          data-srcset="/img/error-page/background_lg.png"
          media="(min-width: 992px)"
          type="image/png"
        />
         
        <source
          data-srcset="/img/error-page/background_md.webp"
          media="(min-width: 768px)"
          type="image/webp"
        />
         
        <source
          data-srcset="/img/error-page/background_md.png"
          media="(min-width: 768px)"
          type="image/png"
        />
         
        <source
          data-srcset="/img/error-page/background.webp"
          type="image/webp"
        />
         
        <img
          alt="hanger"
          class="not-found__image lazyload"
          data-object-fit="cover"
          data-object-position="center center"
          data-src="/img/error-page/background.png"
          height="100"
          loading="lazy"
          width="100"
        />
      </picture>
    </div>
     
    <div
      class="not-found__container container"
    >
      <div
        class="not-found__title"
      >
        
      </div>
       
      <div
        class="not-found__description"
      >
        
      
    
      </div>
       
      <button
        class="btn not-found__link btn-primary"
        type="button"
      >
        
      
    
      </button>
    </div>
  </div>
</main>
`;
